### PR TITLE
Fix testsuite failures and makes more salt features idempotent

### PR DESCRIPTION
**Pull request by meaksh**

### CODE REVIEW

File: features/check_registration.feature [hunk 1]
```diff
@@ -318,7 +291,7 @@ Feature: Check client registration
     Then I should see a "System Snapshots" text
     And I should see a "Package profile changed" link
 
-  Scenario: check tab links "Provisioning" => "Snapshots Rolloback"
```
Comment (MalloZup) [inline features/check_registration.feature]: LOL

File: features/register_client.feature [hunk 2]
```diff
@@ -9,3 +9,31 @@ Feature: Register a client
   Scenario: Register a client
     When I register using an activation key
     Then I should see this client in spacewalk
+
```
Comment (MalloZup) [inline features/register_client.feature]: @meaksh  when we register trad-client, when does it happen the refresh stuff? we can try to run it

File: features/support/twopence_init.rb [hunk 3]
```diff
@@ -78,6 +78,46 @@ def get_target(host)
...
+    hostname = $client_hostname
+  when "sle-migrated-minion"
+    hostname = $client_hostname
+  else
+    raise "Invalid target."
+  end
+  hostname
+end
+
+def get_target_fullhostname(host)
```
Comment (MalloZup) [inline features/support/twopence_init.rb]: not bad 👍

File: run_sets/testsuite.yml [hunk 4]
```diff
@@ -20,6 +20,11 @@
 - features/centos_salt_ssh.feature 
 ## CORE feature end ###
 
+### Features affected by the package problem.
```
Comment (MalloZup) [inline run_sets/testsuite.yml]: OK 🥇

File: features/register_client.feature (hunk 2 above)
Comment (meaksh) [inline features/register_client.feature]: @MalloZup - It should run auto-magically after client registration.

Comment (MalloZup) [inline features/register_client.feature]: LOL

**Pull request merged**